api/app/Models/Pool.php

```diff
@@ -113,12 +114,50 @@ public function publishedPoolCandidates(): HasMany
 
     public function essentialSkills(): BelongsToMany
     {
-        return $this->belongsToMany(Skill::class, 'pools_essential_skills')->withTrashed(); // always fetch all skills for a pool
+        return $this->belongsToMany(Skill::class, 'pool_skill')
+            ->withTrashed() // pool-skills always fetches soft-deleted skill models
+            ->withTimestamps()
+            ->wherePivot('type', PoolSkillType::ESSENTIAL->name);
     }
 
     public function nonessentialSkills(): BelongsToMany
     {
-        return $this->belongsToMany(Skill::class, 'pools_nonessential_skills')->withTrashed();
+        return $this->belongsToMany(Skill::class, 'pool_skill')
+            ->withTrashed()
+            ->withTimestamps()
+            ->wherePivot('type', PoolSkillType::NONESSENTIAL->name);
+    }
+
+    /**
+     * Sync the essential skills in pool_skill
+     *
+     * @param $skills - array of skill ids
+     * @return void
+     */
+    public function setEssentialPoolSkills($skills)
```

Comment:
I'd recommend renaming `$skills` to `$skillIds`, to make it obvious that the whole object shouldn't be passed in. Same for line 153.

Reply:
Aye aye https://github.com/GCTC-NTGC/gc-digital-talent/pull/7953/commits/f8af0abc4da9755572327b418b2c3941f2874036